### PR TITLE
Allow yml configuration

### PR DESCRIPTION
**Pull request by roelywoely**

### CODE REVIEW

**Comment by roelywoely** [comment]
Can you merge it?

**Comment by othillo** [comment]
removing support for phpunit 4.8 broke the build https://github.com/qandidate-labs/qandidate-toggle-bundle/pull/36/files#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780L31

either revert this or wait for https://github.com/qandidate-labs/qandidate-toggle-bundle/pull/37 to get merged.

File: Factory/SymfonyCollection.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+class SymfonyCollection
+{
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * SymfonyCollection constructor.
```
Comment (othillo) [inline Factory/SymfonyCollection.php]: you can remove this line

File: Factory/SymfonyCollection.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * @return \Qandidate\Toggle\ToggleCollection\InMemoryCollection
+     */
+    public function getToggles()
+    {
+        $serializer = new InMemoryCollectionSerializer();
+        return $serializer->deserialize($this->config);
+    }
+}
```
Comment (othillo) [inline Factory/SymfonyCollection.php]: missing newline (PSR-2)

File: QandidateToggleBundle.php [hunk 3]
```diff
@@ -11,8 +11,11 @@
 
 namespace Qandidate\Bundle\ToggleBundle;
 
+use Qandidate\Bundle\ToggleBundle\DependencyInjection\QandidateToggleExtension;
```
Comment (othillo) [inline QandidateToggleBundle.php]: why did you add these lines?

File: phpunit.xml.dist [hunk 4]
```diff
@@ -17,4 +17,14 @@
     </testsuite>
   </testsuites>
 
+  <filter>
```
Comment (othillo) [inline phpunit.xml.dist]: what does this add to the currenct test setup?

File: Factory/SymfonyCollection.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Qandidate\Bundle\ToggleBundle\Factory;
+
+use Qandidate\Toggle\Serializer\InMemoryCollectionSerializer;
+
+/**
+ * Class SymfonyCollection
+ * @package Qandidate\Bundle\ToggleBundle\Factory
+ */
+class SymfonyCollection
```
Comment (othillo) [inline Factory/SymfonyCollection.php]: I think this should be called InMemoryCollectionFactory

File: Factory/SymfonyCollection.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @return \Qandidate\Toggle\ToggleCollection\InMemoryCollection
+     */
+    public function getToggles()
+    {
+        $serializer = new InMemoryCollectionSerializer();
```
Comment (othillo) [inline Factory/SymfonyCollection.php]: please inject the serializer in the constructor

File: DependencyInjection/QandidateToggleExtension.php [hunk 7]
```diff
@@ -53,6 +56,18 @@ public function load(array $configs, ContainerBuilder $container)
                 ));
 
                 $container->setDefinition('qandidate.toggle.collection.factory', $definition);
+
+                break;
+            case 'symfony' === $config['persistence']:
+                $container->setParameter('qandidate.toggle.parameters', $config['toggles']);
+                $collection = 'factory';
+                $definition = new DefinitionDecorator('qandidate.toggle.collection.in_memory');
+                $definition->setFactory(array(
```
Comment (othillo) [inline DependencyInjection/QandidateToggleExtension.php]: not sure, but can't you just use the InMemoryCollectionSerializer as factory?

$definition->setFactory([
    new Reference('qandidate.toggle.serializer.in_memory_collection'), // @todo add to services.xml
   'deserialize',
]);
$definition->setArguments(['%qandidate.toggle.parameters%']);

if this works, you don't have to use the SymfonyCollection.

Comment (roelywoely) [inline DependencyInjection/QandidateToggleExtension.php]: No it doesn't work, I just checked it.

RuntimeException in ResolveDefinitionTemplatesPass.php line 189:
Service "qandidate.toggle.collection.factory": Invalid argument key "extended_users" found.

extended_users is the name of a feature toggle of ours.

**Comment by roelywoely** [comment]
I didn't do the original changes so I needed to look into it.
Now all requested changes have been done. Please let me know if something still should be changed.

File: DependencyInjection/QandidateToggleExtension.php (hunk 7 above)
Comment (othillo) [inline DependencyInjection/QandidateToggleExtension.php]: @roelywoely I got it working in https://github.com/qandidate-labs/qandidate-toggle-bundle/pull/39

**Comment by othillo** [comment]
closed in favor of https://github.com/qandidate-labs/qandidate-toggle-bundle/pull/39
